crates/agent/src/publications/specs.rs

```diff
@@ -279,6 +279,18 @@ pub fn validate_transition(
                 // No constraint.
             }
         };
+
+        // Verify that the live specification has not existed and then been deleted in the past.
+        // TODO(johnny): remove once we introduce data plane pet-names.
+        if live_type.is_none() && *last_pub_id != pub_id {
```

Comment:
So I'm not familiar with this code that much, but reading this vs the stated intent of "disallowing re-creating of a deleted spec", I wonder if there's a missing condition of `&& draft_type.is_some()` here? 

I read what's here as "If the live value has been deleted, and we're creating a new publication id, then disallow because you can't re-create a deleted thing". But is it possible that you'd want to do a new publication, _also_ with a value of None for this draft spec? Maybe because some other draft spec changed, but this one is still deleted...? I don't know enough about how this stuff fits together to know whether this is a concern or not.

Reply:
You're not wrong: as written, the code says "you can't update _or delete_ a live-spec which existed but has since been deleted". Your check would allow for a repeated deletion of an already-deleted live spec, which doesn't really come up but is more logically consistent. I'll add.

Reply:
Done